libgadget/forcetree.c

```diff
@@ -474,7 +475,7 @@ int force_tree_create_nodes(const ForceTree tb, const int npart, DomainDecomp *
     for(i = 0; i < npart; i++)
     {
         /*Can't break from openmp for*/
-        if(nc.nnext_thread >= tb.lastnode-1)
+        if(nc.nnext_thread >= tb.lastnode)
```

Comment:
I don't quite get this. Looks like the previous condition to 'terminate' the loop is less strict, and the 'recovery' branch should have been able to handle it the same.

Reply:
What happened before is that the condition to skip the rest of the particles would hit, and the condition to signal that some particles were skipped would not. Because we started skipping particles before we ran out of memory we would never run out of memory. We would instead be handed an incomplete tree.